app/scripts/controllers/push-platform-notifications/firebase/utils/get-notification-message.ts

```diff
@@ -0,0 +1,285 @@
+// eslint-disable-next-line spaced-comment
+/// <reference lib="WebWorker" />
+import { CHAIN_SYMBOLS } from '../../../../../../shared/constants/platform-notifications';
+import type { TRIGGER_TYPES } from '../../../../../../shared/constants/platform-notifications';
+import type { OnChainRawNotification } from '../../types/on-chain-notification/on-chain-notification';
+import { t } from '../../../../translate';
+import { getAmount, formatAmount } from './get-notification-data';
+
+type PushNotificationMessage = {
+  title: string;
+  description: string;
+};
+
+type ChainSymbols = {
+  [key: string]: string;
+};
+
+type NotificationMessage<
+  N extends OnChainRawNotification = OnChainRawNotification,
+> = {
+  title: string;
+  defaultDescription: string;
+  getDescription?: (n: N) => string | null;
+};
+
+type NotificationMessageDict = {
+  [K in TRIGGER_TYPES]?: NotificationMessage<
+    Extract<OnChainRawNotification, { data: { kind: `${K}` } }>
+  >;
+};
+
+const sw = self as unknown as ServiceWorkerGlobalScope;
+
+function getChainSymbol(chainId: number) {
+  const hexChainId = `0x${chainId.toString(16)}`;
+  return (CHAIN_SYMBOLS as ChainSymbols)[hexChainId] ?? null;
+}
+
+export async function onPushNotification(notification: unknown) {
+  if (!notification) {
+    return;
+  }
+  if (!isOnChainNotification(notification)) {
+    return;
+  }
+
+  const notificationMessage = createNotificationMessage(notification);
+  if (!notificationMessage) {
+    return;
+  }
+
+  const registration = sw?.registration;
+  if (!registration) {
+    return;
+  }
+
+  // const iconUrl = await browser.runtime.getURL('../../images/icon-64.png');
+
+  // eslint-disable-next-line consistent-return
+  return registration.showNotification(notificationMessage.title, {
+    body: notificationMessage.description,
+    icon: './images/icon-64.png',
+    tag: notification?.id,
+    data: notification,
+  });
+}
+
+function isOnChainNotification(n: unknown): n is OnChainRawNotification {
+  const assumed = n as OnChainRawNotification;
+
+  // We don't have a validation/parsing library to check all possible types of an on chain notification
+  // It is safe enough just to check "some" fields, and catch any errors down the line if the shape is bad.
+  const isValidEnoughToBeOnChainNotification = [
+    assumed?.id,
+    assumed?.data,
+    assumed?.trigger_id,
+  ].every((field) => field !== undefined);
+  return isValidEnoughToBeOnChainNotification;
+}
+
+const notificationMessageDict: NotificationMessageDict = {
+  erc20_sent: {
+    title: t('pushPlatformNotificationsFundsSentTitle') ?? 'Funds sent',
```

Comment:
These fallbacks to english strings (e.g. `?? 'Funds sent'`) should not be needed, because the `t` function from the `translate.ts` file falls back to the english translation if no translation in the current locale exists

Reply:
the same applied to all cases in this file

Reply:
Ah ok, perfect!

Reply:
Uhm, now I remember why I had included this fallback. Without it, I get an eslint error "Type 'string | null' is not assignable to type 'string'". I'll try to investigate a bit more!

Reply:
Ok, solved!

Reply:
Yep nice callout!